src/java.base/share/classes/jdk/internal/crac/JDKFileResource.java

```diff
@@ -34,20 +42,44 @@ public void beforeCheckpoint(Context<? extends Resource> context) throws Excepti
             return;
         }
 
-        Supplier<Exception> exceptionSupplier;
-        if (matchClasspath(path)) {
-            // Files on the classpath are considered persistent, exception is not thrown
-            exceptionSupplier = () -> null;
-        } else {
-            exceptionSupplier = () -> new CheckpointOpenFileException(path, getStackTraceHolder());
-        }
-
+        OpenResourcePolicies.Policy policy = OpenResourcePolicies.findForPath(false, path);
+        String action = policy != null ? policy.action.toLowerCase() : "error";
+        Supplier<Exception> exceptionSupplier = switch (action) {
+            case "error":
+                error = true;
+                // Files on the classpath are considered persistent, exception is not thrown
+                yield matchClasspath(path) ? NO_EXCEPTION :
+                        () -> new CheckpointOpenFileException(path, getStackTraceHolder());
```

Comment:
So error + no exception is a valid combination? Should not we choose `ignore` as the default for files matching classpath instead?